src/python/pants/invalidation/cache_manager.py

```diff
@@ -172,20 +172,34 @@ def __init__(self, cache_manager, target, cache_key):
     super(VersionedTarget, self).__init__(cache_manager, [self])
     self.id = target.id
 
+  def _results_dir_prefix(self, root_dir):
```

Comment:
This method having sideeffects makes me a bit uncomfortable. Could this logic move under `create_results_dir`?

But it is also being called once per target. Best might be for it to move into a static method called during setup by the cachemanager so that each target isn't executing it.